frontend/src/js/external-forms/transformQueryToApi.ts

```diff
@@ -1,21 +1,23 @@
 import { transformElementsToApi } from "../api/apiHelper";
 import type { SelectOptionT } from "../api/types";
 import type { DateStringMinMax } from "../common/helpers";
+import { exists } from "../common/helpers/exists";
 import type { DragItemQuery } from "../standard-query-editor/types";
 
 import type { Form, GeneralField } from "./config-types";
+import type { FormConceptGroupT } from "./form-concept-group/formConceptGroupState";
 import type { DynamicFormValues } from "./form/Form";
 import { isFormField } from "./helper";
 
-function transformElementGroupsToApi(elementGroups) {
+function transformElementGroupsToApi(elementGroups: FormConceptGroupT[]) {
   return elementGroups.map(({ concepts, connector, ...rest }) =>
     concepts.length > 1
       ? {
           type: connector,
-          children: transformElementsToApi(concepts),
+          children: transformElementsToApi(concepts.filter(exists)),
           ...rest,
         }
-      : { ...transformElementsToApi(concepts)[0], ...rest },
+      : { ...transformElementsToApi(concepts.filter(exists))[0], ...rest },
```

Comment:
Couldn't this cause an Index out of Bounds? Or is it ensured that you'll have existing concepts

Reply:
Done!

Reply:
Could have actually led to a bug if you tried to only fill the second or-condition of an otherwise optional concept field. So good catch!